conans/model/ref.py

```diff
@@ -20,23 +32,33 @@ def invalid_name_message(name):
                        % (name, ConanName._min_chars))
         else:
             message = ("'%s' is an invalid name. Valid names MUST begin with a "
-                       "letter or number, have between %s-%s chars, including "
+                       "letter, number or underscore, have between %s-%s chars, including "
                        "letters, numbers, underscore, dot and dash"
                        % (name, ConanName._min_chars, ConanName._max_chars))
         raise InvalidNameException(message)
 
+    @staticmethod
+    def validate_string(value, message=None):
+        """Check for string"""
+        if not isstr(value):
+            message = message or "'{value}' {reason}"
+            raise InvalidNameException(message.format(value=value, reason="is not an string"))
+
     @staticmethod
     def validate_user(username):
+        ConanName.validate_string(username)
         if ConanName._validation_pattern.match(username) is None:
             ConanName.invalid_name_message(username)
 
     @staticmethod
     def validate_name(name, version=False):
         """Check for name compliance with pattern rules"""
+        ConanName.validate_string(name)
         if name == "*":
             return
         if ConanName._validation_pattern.match(name) is None:
             if version and name.startswith("[") and name.endswith("]"):
+                # TODO: Check value between brackets
```

Comment:
Leave it as TODO or check version ranges according to semver as documented (https://docs.conan.io/en/latest/mastering/version_ranges.html)?

Reply:
Could be done in a different PR, not urgent, leave it.